editor/editor_tool_drawer.cpp

```diff
@@ -0,0 +1,496 @@
+/*************************************************************************/
+/*  editor_tool_drawer.cpp                                               */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "editor_tool_drawer.h"
+
+#include "editor_scale.h"
+
+void EditorToolDrawerSection::set_title(const String p_title) {
+	title_label->set_text(p_title);
+}
+
+void EditorToolDrawerSection::set_title_icon(const Ref<Texture2D> &p_icon) {
+	title_icon->set_texture(p_icon);
+}
+
+void EditorToolDrawerSection::add_content(Control *p_content) {
+	ERR_FAIL_COND_MSG(!p_content, "EditorToolDrawerSection content must be a Control node.");
```

Comment:
```suggestion
	ERR_FAIL_COND_MSG(!p_content, "EditorToolDrawerSection content must be a valid Control node.");
```
Also the macro can be changed to `ERR_FAIL_NULL_MSG`.